File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def chtoM(s):
    h,m=int(s[0:2]),int(s[3:5])
    return h*60+m

def ToStr(mm):
	hh, mm = mm //60, mm % 60
	if (hh <10):
		ss = "0"+str(hh)+":"
	else:
		ss = str(hh)+":"
	if (mm < 10):
		ss += "0"+str(mm)
	else:
		ss += str(mm)
	return ss

def Check(tar, st, ed):
	if ((tar >=st) and (tar <=ed)):
		return ed
	else:
		return tar
	

def solution(video_len, pos, op_start, op_end, commands):
	vl=chtoM(video_len)
	pos=chtoM(pos)
	ops=chtoM(op_start)
	ope=chtoM(op_end)
	pos=Check(pos,ops,ope)
	for i in commands:
		if i=="next":
			pos=min(vl,pos+10)
		else:
			pos=max(0,pos-10)
		pos=Check(pos,ops,ope)
	return ToStr(pos)

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    
    static int ttt(String s){
        String[] aa = s.split(":");
        int k = Integer.parseInt(aa[0])*60+Integer.parseInt(aa[1]);
        return k;
    }
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        int vl=ttt(video_len),p=ttt(pos),os=ttt(op_start),oe=ttt(op_end);
        if(p>=os && p<oe)
                p=oe;
        for(String i : commands){
            if(i.equals("prev"))
                p-=10;
            else if (i.equals("next"))
                p+=10;
            if (p<0)
                p=0;
            if(p>vl)
                p=vl;
            if(p>=os && p<oe)
                p=oe;
            
        }
        
        String answer = "";
        return String.format("%02d:%02d",p/60,p%60)
            ;
    }
    
}